qt/scientific_interfaces/test/ISISReflectometry/MainWindow/MainWindowPresenterTest.h

```diff
@@ -162,6 +163,41 @@ class MainWindowPresenterTest : public CxxTest::TestSuite {
     verifyAndClear();
   }
 
+  void testWarningGivenIfRemoveUnsavedBatch() {
+    auto optionsDialogPresenter =
+        std::make_unique<NiceMock<MockOptionsDialogPresenter>>();
+    ON_CALL(*optionsDialogPresenter.get(),
+            getBoolOption(std::string("WarnDiscardChanges")))
+        .WillByDefault(Return(true));
+    auto presenter = makePresenter(std::move(optionsDialogPresenter));
+    auto const batchIndex = 0;
+    ON_CALL(*m_batchPresenters[batchIndex], isBatchUnsaved())
```

Comment:
Could these `ON_CALL` be `EXPECT_CALL` instead? And could they be moved to helper functions to improve readability? e.g. something like `expectWarnDiscardChanges()` and `expectBatchUnsaved()` .... actually it looks like the latter already exists and is being called below so I think the `ON_CALL` is spurious in this case!